### PR TITLE
Add configurations for account verification on email address update.

### PR DESCRIPTION
**Pull request by dewniMW**
### Proposed changes in this pull request

- Adding server-wide configuration to enable and configure related properties involving email address verification on update functionality.
- Subscribing PRE_SET_USER_CLAIMS and POST_SET_USER_CLAIMS events with user email verification handler.
- Introducing a claim to store the user's email address to be updated until account verification happens.

This PR is related to feature: https://github.com/wso2/product-is/issues/7183

### When should this PR be merged
This PR should be merged after https://github.com/wso2-extensions/identity-userstore-ldap/pull/34 and https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/232

### CODE REVIEW

File: features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml [hunk 1]
```diff
@@ -655,6 +655,14 @@
 				<AttributeID>uuid</AttributeID>
 				<Description>Claim to store the user's unique id. (UUID)</Description>
 			</Claim>
+			<Claim>
+				<ClaimURI>http://wso2.org/claims/identity/verificationPendingEmail</ClaimURI>
+				<DisplayName>Verification Pending Email</DisplayName>
+				<!-- If user store based identity store is used, a proper mapped attribute
+				in your user store must be configured for this. -->
+				<AttributeID>verificationPendingEmail</AttributeID>
+				<Description>Claim to store email address to be updated until account verification.</Description>
```
Comment (somindatommy) [inline features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml]: A user's account is already verified at this stage. So here user tries to verify the channel. So I think this should be like `Claim to store newly updated email address until the new email address is verified`. WDYT?

Comment (dewniMW) [reply features/claim-mgt/org.wso2.carbon.claim.mgt.server.feature/resources/conf/claim-config.xml]: +1
Changed it accordingly

**Comment by dewniMW** [comment]
This PR is dependent on https://github.com/wso2-extensions/identity-governance/pull/349. Therefore don’t merge this PR till https://github.com/wso2-extensions/identity-governance/pull/349 is merged.

**Comment by somindatommy** [comment]
Please update the PR description with following details
 - When to merge
 - Follow up actions

**Pull request merged**